cmd/inspector/main.go

```diff
@@ -435,9 +449,106 @@ func CreateCSVStats(cmd *cobra.Command, args []string) (err error) {
 	return nil
 }
 
+func getSegments(cmd *cobra.Command, args []string) error {
+	if irreparableLimit <= int32(0) {
+		return ErrArgs.New("limit must be greater than 0")
+	}
+
+	i, err := NewInspector(*Addr, *IdentityPath)
+	if err != nil {
+		return ErrInspectorDial.Wrap(err)
+	}
+
+	length := irreparableLimit
+	var offset int32
+
+	// query DB and paginate results
+	for length >= irreparableLimit {
+		res, err := i.irrdbclient.ListSegments(context.Background(), &pb.ListSegmentsRequest{Limit: irreparableLimit, Offset: offset})
+		if err != nil {
+			return ErrRequest.Wrap(err)
+		}
+
+		segments, err := unpackSegmentDetail(res.Segments)
+		if err != nil {
+			return err
+		}
+
+		objects := sortSegments(segments)
+		if len(objects) == 0 {
+			break
+		}
+		// format and print segments
+		result, err := json.Marshal(objects)
+		if err != nil {
+			return err
+		}
+
+		var out bytes.Buffer
+		err = json.Indent(&out, result, "", "  ")
+		if err != nil {
+			return err
+		}
+		_, err = out.WriteTo(os.Stdout)
+		if err != nil {
+			return err
+		}
+
+		length = int32(len(res.Segments))
+		offset += length
+
+		if length >= irreparableLimit {
+			if !prompt.Confirm("\nNext page? (y/n)") {
+				break
+			}
+		}
+	}
+	return nil
+}
+
+type formattedSegment struct {
+	EncryptedSegmentPath string      `json:"encrypted_segment_path"`
+	LostPieces           int32       `json:"lost_pieces"`
+	LastRepairAttempt    time.Time   `json:"last_repair_attempt"`
+	RepairAttemptCount   int64       `json:"repair_attempt_count"`
+	SegmentDetail        *pb.Pointer `json:"segment_detail"`
+}
+
+// sortSegments by the object they belong to
+func sortSegments(segments []*formattedSegment) map[string][]*formattedSegment {
+	objects := make(map[string][]*formattedSegment)
+	for _, seg := range segments {
+		path := storj.SplitPath(seg.EncryptedSegmentPath)
+		path = append(path[:1], path[2:]...)
+		objPath := strings.Join(path, "/")
+		objects[objPath] = append(objects[objPath], seg)
+	}
+	return objects
+}
+
+// unpackSegmentDetail unmarshals a slice of segments' pointers for viewing
+func unpackSegmentDetail(packed []*pb.IrreparableSegment) (unpacked []*formattedSegment, err error) {
+	for _, seg := range packed {
+		p := &pb.Pointer{}
+		err = proto.Unmarshal(seg.SegmentDetail, p)
```

Comment:
This isn't your fault, but its a little bogus that EncryptedSegmentDetail is a *marshaled* pb.Pointer and that EncryptedSegmentPath is a search prefix / storage.Key.  Neither is actually encrypted.  There's likely no coherent reason that you have to manually unmarshal SegmentDetail.  I give you permission to rage refactor this.  :D :P